### PR TITLE
Update honeywell documentation

### PR DESCRIPTION
**Pull request by rdfurman**
## Proposed change

I have converted the Honeywell Total Connect Comfort (US) climate platform into a full integration with climate and sensor devices. This is because the Honeywell TCC (US) system actually supports additional sensors (like outdoor sensors). This PR is to update the documentation on home-assistant.io to reflect the changes I made to the integration (separate PR to follow).

<details>
  <summary>Screenshot</summary>
  
  honeywell website
</details>

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/50731

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @rdfurman, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#17814).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by rdfurman** [comment]
Added link to integration change PR. https://github.com/home-assistant/core/pull/50547

**Comment by rdfurman** [comment]
The new look:
image

File: source/_integrations/honeywell.markdown [hunk 1]
```diff
@@ -3,30 +3,27 @@ title: Honeywell Total Connect Comfort (US)
...
+ha_codeowners:
+  - '@rdfurman'
 ha_domain: honeywell
 ha_platforms:
   - climate
+  - sensor
 ---
 
-The `honeywell` climate platform integrates Home Assistant with _US-based_ [Honeywell Total Connect Comfort (TCC)](https://mytotalconnectcomfort.com/portal/) climate systems.
+The `honeywell` integration integrates Home Assistant with _US-based_ [Honeywell Total Connect Comfort (TCC)](https://mytotalconnectcomfort.com/portal/) climate systems.
```
Comment (frenck) [inline source/_integrations/honeywell.markdown]:
```suggestion
The Honeywell integration integrates Home Assistant with _US-based_ [Honeywell Total Connect Comfort (TCC)](https://mytotalconnectcomfort.com/portal/) climate systems.
```

File: source/_integrations/honeywell.markdown [hunk 2]
```diff
@@ -3,30 +3,27 @@ title: Honeywell Total Connect Comfort (US)
...
+{% include integrations/config_flow.md %}
 
-```yaml
-climate:
-# Example configuration.yaml entry
-  - platform: honeywell
-    username: YOUR_USERNAME
-    password: YOUR_PASSWORD
-```
+## Configuration
```
Comment (frenck) [inline source/_integrations/honeywell.markdown]: This whole section, including all configuration variables can be removed as integration using the configuration flow, should deprecated YAML support.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/honeywell.markdown [hunk 3]
```diff
@@ -3,30 +3,27 @@ title: Honeywell Total Connect Comfort (US)
 description: Instructions on how to integrate Honeywell thermostats within Home Assistant.
 ha_category:
   - Climate
+  - Sensor
```
Comment (frenck) [inline source/_integrations/honeywell.markdown]:
```suggestion
```

File: source/_integrations/honeywell.markdown [hunk 4]
```diff
@@ -3,30 +3,27 @@ title: Honeywell Total Connect Comfort (US)
...
 ha_release: pre 0.7
 ha_iot_class: Cloud Polling
+ha_config_flow: true
+ha_codeowners:
+  - '@rdfurman'
 ha_domain: honeywell
 ha_platforms:
   - climate
+  - sensor
```
Comment (frenck) [inline source/_integrations/honeywell.markdown]:
```suggestion
```

**Comment by rdfurman** [comment]
new-honeywell-docs

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Pull request merged**